### ISSUE TITLE
Double Elimination save option

### ISSUE THREAD

**Issue by thecodeassassin** [teijo/jquery-bracket#36]
Hey there,

I'm using the saveFn function to save results to the database. The problem i'm having is that this doesn't work with Double Elimination since detecting whether or not it's double elimination depends on the data itself. Is there a flag i can use to force it to double elimination?

**Comment by teijo** [comment]
`Is there a flag i can use to force it to double elimination?`

No there isn't. You can force the double elimination by initializing with empty data structure that has room for loser bracket and final data.

**Comment by thecodeassassin** [comment]
That doesn't work. The saving process makes it into a single elimination tournament:

http://fiddle.jshell.net/7er6y8j7/3/

**Comment by teijo** [comment]
Fiddling with the save demo at http://www.aropupu.fi/bracket/#save. The output seems to contain empty data for double elimination results when saved (name/score edited). It just doesn't seem to trigger if bracket size or type is changed from the buttons.

Didn't try importing the empty data created by the bracket, but if the data import worked, then the fix would probably be triggering `saveFn` on size and type change to avoid confusion.

**Comment by thecodeassassin** [comment]
Hey teijo,

can you check this line ? https://github.com/teijo/jquery-bracket/blob/master/src/jquery.bracket.ts#L1109

i could create a PR to make this an optional setting, would that work?

**Comment by thecodeassassin** [comment]
PR made: https://github.com/teijo/jquery-bracket/pull/38

**Comment by teijo** [comment]
**tl;dr** Bracket type still depends on initialization data, and it should. Unfortunately the double elimination initialization data is ugly if backwards compatibility is concerned: `[[[[]]], [], []]`.

I went through the input handling and found what the problem is. From https://github.com/teijo/jquery-bracket/blob/3a4210cb8bca6b44f2789d12197c5f5815491a42/src/jquery.bracket.ts#L1103

```javascript
    var r = data.results

    /* wrap data to into necessary arrays */
    r = wrap(r, 4 - depth(r))
    data.results = r

    var isSingleElimination = (r.length <= 1)
```

The results data is normalized with the `wrap()`. It nests arrays until they are 4 deep, that is:
 1. list of brackets
 2. list of rounds in bracket
 3. list of matches in round
 4. a match

If nesting is missing, it will wrap the data until 4 levels exist, therefore giving just `[1, 2]` as result would convert it as result of first match of first round in single bracket -> `[[[[1, 2]]]]`.

The bracket type is determined by the number of bracket lists (top level) in the normalized result input. 1 for single bracket, greater than 1 for double elimination. Example above in double elimination would be `[[[[1, 2]]], [], []]`

Closing this issue as the data can define the bracket type. As the bracket edit mode output by `saveFn` will output normalized data which, when read in init, will result into correct behaviour, no additional option for double elimination output is needed.

Improvement to the API would be to discard the normalization and enforce exact input format from the beginning. However, that will not be backwards compatible and therefore not desirable at this point.
